src/utilities/convertDollarsToCents.ts

```diff
@@ -1,6 +1,14 @@
 import BigNumber from 'bignumber.js';
 
-const convertDollarsToCents = (value: number | BigNumber) =>
-  new BigNumber(value).times(100).dp(0).toNumber();
+const convertDollarsToCents = (
+  value: number | BigNumber,
+  { whole }: { whole: boolean } = { whole: true },
+) => {
+  const cents = new BigNumber(value).times(100);
+  if (whole) {
+    return cents.dp(0).toNumber();
+  }
+  return cents.toNumber();
+};
```

Comment:
I'd remove the option to return rounded cents and always return fractional cents to keep things simple. The dApp will still only display 2 decimals for dollar values since `formatCentsToReadableValue` takes care of that.